### PR TITLE
Update feature_alignment (now TRIC)

### PR DESCRIPTION
**Pull request by npinter**
This PR updates feature_alignment which was renamed to TRIC.

### CODE REVIEW

File: tools/feature_alignment/feature_alignment.xml [hunk 1]
```diff
@@ -1,168 +1,288 @@
...
     </requirements>
-    <stdio>
-        <!-- Anything other than zero is an error -->
-        <exit_code range="1:" />
-        <exit_code range=":-1" />
-        <!-- In case the return code has not been set propery check stderr too -->
-        <regex match="Error:" />
-        <regex match="Exception:" />
-    </stdio>
-    <version_command>feature_alignment.py --version</version_command>
```
Comment (bgruening) [inline tools/feature_alignment/feature_alignment.xml:15]: version command would be nice to keep

File: tools/feature_alignment/feature_alignment.xml [hunk 2]
```diff
@@ -1,168 +1,288 @@
...
-    <command>
-<![CDATA[
+    <command detect_errors="exit_code"><![CDATA[
     feature_alignment.py 
-        --in #echo '" "'.join([ str($file) for $file in $inputs ]) #
-        --file_format $iformat
-        ##if str($iformat) == "openswath":
-            --out $out
-        ##end if
+        --in #echo '" "'.join([ str($file) for $file in $inputs ])
```
Comment (bgruening) [inline tools/feature_alignment/feature_alignment.xml:14]: single quotes needs to be the one on the commandline, not the double quotes

File: tools/feature_alignment/feature_alignment.xml [hunk 3]
```diff
@@ -1,168 +1,288 @@
...
         #end if
-        --method $method
-        --matrix_output_method $matrix_output_method
-        --realign_method $realign_method
-        &&
-        mv out_matrix.tsv $out_matrix
 
-]]></command>
+    #if $iformat == "openswath":
+        && cat out.tsv > $out
```
Comment (bgruening) [inline tools/feature_alignment/feature_alignment.xml:63]: single quotes

File: tools/feature_alignment/feature_alignment.xml [hunk 4]
```diff
@@ -1,168 +1,288 @@
...
+            --target_fdr "0"
+        #end if
+        --max_fdr_quality $max_fdr_quality
+        #if $method_sel.method == "LocalMST" or $method_sel.method == "LocalMSTAllCluster":
+            --mst:useRTCorrection $method_sel.mst_useRTCorrection
+            --mst:Stdev_multiplier $method_sel.mst_Stdev_multiplier
+            --mst:useLocalStdev $method_sel.mst_useLocalStdev
+            --mst:useReference $method_sel.mst_useReference
+        #end if
+        #if $further_params.disable_isotopic_grouping == "True":
```
Comment (bgruening) [inline tools/feature_alignment/feature_alignment.xml:45]:
```suggestion
        #if $further_params.disable_isotopic_grouping:
```

File: tools/feature_alignment/feature_alignment.xml [hunk 5]
```diff
@@ -1,168 +1,289 @@
-<tool id="feature_alignment" name="Feature Alignment" version="0.1">
-    <description>of peakgroups below a FDR</description>
+<tool id="feature_alignment" name="TRIC" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@">
+    <description>integrates information from all available runs via a graph-based alignment strategy</description>
+     <macros>
+        <token name="@TOOL_VERSION@">0.11.0</token>
+        <token name="@GALAXY_VERSION@">0</token>
+         <import>macros.xml</import>
```
Comment (bgruening) [inline tools/feature_alignment/feature_alignment.xml:6]: indentation

File: tools/feature_alignment/feature_alignment.xml [hunk 6]
```diff
@@ -1,168 +1,289 @@
...
         #end if
-        --method $method
-        --matrix_output_method $matrix_output_method
-        --realign_method $realign_method
-        &&
-        mv out_matrix.tsv $out_matrix
 
-]]></command>
+    #if $iformat == 'openswath':
+        && cat out.tsv > $out
```
Comment (bgruening) [inline tools/feature_alignment/feature_alignment.xml:64]: single quotes :)

File: tools/feature_alignment/feature_alignment.xml [hunk 7]
```diff
@@ -1,168 +1,289 @@
...
+    #else:
+        && echo '' > $out
+    #end if
+    && cat out_matrix.tsv > $out_matrix
+    && cat out_ids.tsv > $out_ids
```
Comment (bgruening) [inline tools/feature_alignment/feature_alignment.xml:69]: and here and above

**Review by bgruening — COMMENTED** [review]
tests are failing now :(

**Comment by npinter** [comment]
> tests are failing now :(

I can't find the 'All tool test results' artfiact?!

**Comment by bgruening** [comment]
grafik

**Comment by bgruening** [comment]
under summary :)

**Comment by npinter** [comment]
> under summary :)

thank you! I was expecting it somewhere here :) (it was there before):
grafik

The error is:
`Job in error state.. tool_id: feature_alignment, exit_code: 127, stderr: /tmp/tmp1zbnmvo3/job_working_directory/000/2/tool_script.sh: line 9: feature_alignment.py: command not found`

Requirements are correct:
`<requirement type="package" version="0.11.0">msproteomicstools</requirement>`

https://bioconda.github.io/recipes/msproteomicstools/README.html

Any ideas?

**Comment by bgruening** [comment]
Have you tested this with `planemo test --docker` locally? It seems that the created container does not have the python file?

**Comment by npinter** [comment]
I've tested it with:
`planemo test --no_dependency_resolution`
--> all tests green

and with:
`planemo test --docker`
--> 3 of 5 tests green
--> `feature_alignment.py` is availiable in all tests!

**Comment by bgruening** [comment]
The one with --docker you need to get all green. That is also what is here running.

**Comment by bgruening** [comment]
I restarted the tests.

**Comment by npinter** [comment]
ping @bgruening :)

File: tools/feature_alignment/feature_alignment.xml [hunk 8]
```diff
@@ -1,168 +1,289 @@
...
-        --matrix_output_method $matrix_output_method
-        --realign_method $realign_method
-        &&
-        mv out_matrix.tsv $out_matrix
 
-]]></command>
+    #if $iformat == 'openswath':
+        && cat out.tsv > '$out'
+    #else:
+        && echo '' > '$out'
```
Comment (bgruening) [inline tools/feature_alignment/feature_alignment.xml:66]: if this output is empty it should not be put into the history, left?

Comment (npinter) [reply tools/feature_alignment/feature_alignment.xml]: Thanks for pointing that out!

**Comment by npinter** [comment]
` WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fa756ad0f10>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /packages/34/39/cde2c8a227abb4f9ce62fe55586b920f438f1d2903a1a22514d0b982c333/flake8-4.0.1-py2.py3-none-any.whl`

does the test restarts itself?

**Comment by bgruening** [comment]
Thanks!

**Pull request merged**